tools/typescript_fixes.js

```diff
@@ -0,0 +1,166 @@
+const { readdirSync, readFileSync, writeFileSync, realpathSync, statSync } = require('fs');
+const path = require('path');
+
+console.debugFixLog = false ? console.debug : () => {};
+console.debugReadTypes = false ? console.debug : () => {};
+
+console.log('Currect directory:', realpathSync('./'))
+const typesPath = './types'
+const paths = {
+    'index.esm': path.join(typesPath, 'index.esm.d.ts'),
+};
+
+console.log('Processing', paths['index.esm']);
+
+/**
+ * Removes `namespace log` and adds appropriate import and reference to index.esm.d.ts.
+ *
+ * @param {string[]} input
+ * @return {string[]}
+ */
+const fixLog = (input) => {
+    const output = [];
+
+    const modes = {
+        base: 0, // Most reading and writing is in `base` mode: keep each line as is.
+        'log-ns-replace': 1, // Erasing log namespace declaration and adding proper import
+    };
+
+    let index = 0
+    let mode = modes.base;
+    for (const line of input) {
+        try {
+            if (mode === modes.base) {
+                if (index === 0) {
+                    output.push('/// <reference path="../src/utils_log.d.ts" />');
+                    console.debugFixLog('line #1', mode);
+                }
+                if (line.match(/^declare namespace log {/)) {
+                    mode = modes['log-ns-replace'];
+                    console.debugFixLog('switch to', mode, line);
+                } else {
+                    output.push(line);
+                    console.debugFixLog('out', line);
+                }
+            } else if (mode === modes['log-ns-replace']) {
+                if (line.match(/^\}/)) {
+                    output.push("import * as log from 'apify-shared/log';");
```

Comment:
since it's referencing the utils_log.d.ts types by path at the top, you can get rid of this line entirely (erase both `declare namespace log` and  `import * as log from 'apify-shared/log';`. typeof will show the correct definition

Reply:
Done.